radio/src/gui/colorlcd/model_inputs.cpp

```diff
@@ -343,26 +358,27 @@ void ModelInputsPage::newInput()
   menu->setTitle(STR_MENU_INPUTS);
 
   uint8_t chn = 0;
+  uint8_t index = 0;
   ExpoData* line = g_model.expoData;
 
   // search for unused channels
-  for (uint8_t i = 0; i < MAX_EXPOS; i++) {
+  for (uint8_t i = 0; i < MAX_EXPOS && chn < MAX_INPUTS; i++) {
     if (!EXPO_VALID(line) || (line->chn > chn)) {
-      if (chn >= MAX_INPUTS) break;
-      std::string name(getSourceString(chn+1));
-      menu->addLineBuffered(name.c_str(), [=]() { insertInput(chn, i); });
+      std::string name(getSourceString(chn + 1));
+      menu->addLineBuffered(name.c_str(), [=]() { insertInput(chn, index); });
     }
-    if (EXPO_VALID(line))
+    if (EXPO_VALID(line)) {
       chn = line->chn + 1;
-    else
+      index += 1;
+    } else {
       chn += 1;
+    }
     ++line;
   }
```

Comment:
Is this the section / all that needs changing (so I can pull it into 2.9 for a future release)? It's hard to tell exactly what needs changing as the formatting was done in the same commit.

Reply:
Yes, that's the fix. Sorry for making things harder for you.

Reply:
Thanks for that. 👍 Just in future, if you think there is a good chance a fix might get back-ported, can you do the formatting in a separate commit... just makes it less likely half a fix will get pulled in (especially if needing to work around changes on both sides). 🙂